libs/validation-constraints/src/lib/constraints/coscrad-constraint.enum.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * The values here are meant to be human-readable labels and should complete the
+ * sentence "Must be a valid _____"
+ */
+export enum CoscradConstraint {
+    isNonEmptyString = 'non-empty string',
+    // i.e. not `null` or `undefined`
+    isDefined = 'defined',
```

Comment:
This might actually be a nice constraint function to use instead of `isNullOrUndefined`.

Reply:
How would it catch null values?

Reply:
if(!isDefined(input)){
   // exceptional logic
}

Reply:
That said, sometimes `""`, `[]`, or `{}` represent the absence of a value, so it could cause confusion. `isNullOrUndefined` is more explicit, albeit imperative.